gnomad/utils/vep.py

```diff
@@ -478,32 +478,43 @@ def get_csq_from_struct(
 
         # Add exception for transcripts
         if feature_type == "Transcript":
-            fields.update(
-                {
-                    "canonical": hl.if_else(element.canonical == 1, "YES", ""),
-                    "ensp": element.protein_id,
-                    "gene": element.gene_id,
-                    "symbol": element.gene_symbol,
-                    "symbol_source": element.gene_symbol_source,
-                    "cdna_position": hl.str(element.cdna_start) + hl.if_else(
-                        element.cdna_start == element.cdna_end,
-                        "",
-                        "-" + hl.str(element.cdna_end),
-                    ),
-                    "cds_position": hl.str(element.cds_start) + hl.if_else(
-                        element.cds_start == element.cds_end,
-                        "",
-                        "-" + hl.str(element.cds_end),
-                    ),
-                    "mirna": hl.delimit(element.mirna, "&"),
-                    "protein_position": hl.str(element.protein_start) + hl.if_else(
-                        element.protein_start == element.protein_end,
-                        "",
-                        "-" + hl.str(element.protein_end),
-                    ),
-                    "uniprot_isoform": hl.delimit(element.uniprot_isoform, "&"),
-                }
-            )
+            transcript_dict = {
+                "canonical": hl.if_else(element.canonical == 1, "YES", ""),
+                "ensp": element.protein_id,
+                "gene": element.gene_id,
+                "symbol": element.gene_symbol,
+                "symbol_source": element.gene_symbol_source,
+                "cdna_position": hl.str(element.cdna_start) + hl.if_else(
+                    element.cdna_start == element.cdna_end,
+                    "",
+                    "-" + hl.str(element.cdna_end),
+                ),
+                "cds_position": hl.str(element.cds_start) + hl.if_else(
+                    element.cds_start == element.cds_end,
+                    "",
+                    "-" + hl.str(element.cds_end),
+                ),
+                "mirna": hl.delimit(element.mirna, "&") if "mirna" in element else None,
+                "protein_position": hl.str(element.protein_start) + hl.if_else(
+                    element.protein_start == element.protein_end,
+                    "",
+                    "-" + hl.str(element.protein_end),
+                ),
+                "uniprot_isoform": (
+                    hl.delimit(element.uniprot_isoform, "&")
+                    if "uniprot_isoform" in element
```

Comment:
How about those all-missing fields? Like `swissprot, trembl, uniparc`? And the new fields: `mane_select, mane_plus_clinical, mirna`?

Reply:
I know `uniprot_isoform` is new.

Reply:
i don't think the other fields are impacted